Code Examples/week4/bridge-vue/README.md

```diff
@@ -1232,3 +1232,152 @@ These techniques can be used in your own applications and components.  Also, now
 that you have a better understanding of how to build component based apps, you can
 start exploring using the many third-party Vue components that exist.  Whether you
 write it, or someone else does, the ideas are the same.
+
+
+## Search Bar 
+
+It is an optional augmentation, granted we mind minimal use case for the application.
+![Screencast of SearchBar in Action](screenshots/search-bar.gif)
+
+[Semantic-UI-Vue](https://github.com/Semantic-UI-Vue/Semantic-UI-Vue) is a [Semantic-UI](https://semantic-ui.com/) integration, while Semantic is a flexbox friendly UI framework designed for theming. It will bring about pretty elements for input field, loader and error-modal for `bridge-vue` application.
+
+### Implementation
+It is as simple as:
+- install dependencies
+- include the Semantic JS, CSS files
+- apply Semantic middleware to Vue engine
+- rework BridgeMenu component
+
+To install dependencies:
+```
+npm install semantic-ui-vue semantic-ui-css --save
+```
+
+Then, to make Vue understand Semantic's custom classes and elements, stylesheet and JS module files are imported into `main.js`:
+```
+import SuiVue from 'semantic-ui-vue';
+Vue.use(SuiVue);
+import 'semantic-ui-css/semantic.min.css';
+```
+
+<strong>Template markup</strong> of `BridgeMenu.vue` component is modified accordingly. 
+
+First, outer container `<div class='main'>` is now <em>classified</em>, instead of <em>identified</em>,  as `menu`, for the reason of CSS rules to be reused. 
+
+In this markup, conditional directives `v-if`, `v-else` replace `v-bind`ing of class-attributes (i.e. `active`) that Semantic uses for the sake of readability. 
+
+Lastly, when menu-items are being deployed into the listing, the source array is changed to a new stateful property `this.showBridges`, that is a filtered version of `this.bridges`. 
+
+```angular2html
+<template>
+    <div class="menu">
+        <!--Search Field-->
+        <div style="width:100%;" class="ui icon input">
```

Comment:
What do you think about moving this inline style to the `<styles>` block and adding a class instead, to keep the approach consistent here?

Reply:
Grace to `<style scoped>`, `width` can be safely assigned to `input` class